### DISCUSSION TITLE
Max depth in output does not correspond with minimum depth parameter

### DISCUSSION THREAD

**Discussion by AnneliesBroeckx** [opengeos/lidar#18] [Q&A]
Hi,

Many thanks to develop this package to detect nested depressions based on a DEM!
I have tested this out with a DEM (resolution 5 m) and height in cm. These were my parameters:
- min_size = 400  
- min_depth = 100  
- slicing interval = 20 

So I have set the minimum depth parameter equal to 100 cm. When I run the script in Python and view the output of the depressions_info.csv file, in order to examine all the detected depressions, I notice that the meximum depth of each polygon does not always correspond with the set minimum depth parameter of 100 cm. Many depressions (376 out of 452) have a maximum depth that is smaller than 100 cm. How is this possible? 

I have used the example code (lidar-master\lidar\example.py).

Many thanks for your anwser!

**Comment by giswqs** [comment]
For 100 cm (1m), it should probably be `min_depth = 1`

**Reply by AnneliesBroeckx** [reply]
No, cause the DEM is in centimeters. 
I also had done tests with DEM in meter. But then I get similar results (max depth is smaller than set min depth parameter).
Do you know what can be wrong?

**Comment by giswqs** [comment]
Keep in mind that `min_depth` will only affect the level-1 depressions. Could you check the out csv and see what levels of those depression with `max_depth` smaller than `min_depth`?

**Reply by AnneliesBroeckx** [reply]
I checked, most of them are level 1-depressions! Do you know what might be cause this error?

**Comment by giswqs** [comment]
Could you share a small subset of your dataset for debugging?

**Reply by AnneliesBroeckx** [reply]
I tried to attach here a tif-file, but that filetype is not supported. Then I tried to attach a zip-file, but that didn't seem to work either ("is not included in the list"). How can I send you a DEM tif file?

**Reply by giswqs** [reply]
Upload to Google Drive and share a link here.

**Reply by AnneliesBroeckx** [reply]
I have made a subset of my DEM file (see this [link](https://drive.google.com/file/d/19UTR-Aw50WMLPyiX2fdQjUJaapgFurEX/view?usp=sharing)). The unit is meters.
Many thanks in advance to check!

**Comment by giswqs** [comment]
This seems to be a bug. The `min_depth` parameter has no effect. I will look into it.

**Reply by AnneliesBroeckx** [reply]
Thank you very much! I look forward to your answer.
